src/compiler/utilities.ts

```diff
@@ -3298,6 +3297,10 @@ namespace ts {
         return false;
     }
 
+    export function isJSDocTypeReference(node: TypeReferenceType): node is TypeReferenceNode {

```

Comment:
This is going to be super slow if we're not in jsdoc -- for any type reference *not* in jsdoc we will climb the whole tree looking for a JSDocTypeExpression that never comes.